example/z-docker/Dockerfile

```diff
@@ -1,4 +1,40 @@
-FROM ubuntu:latest
-RUN apt update && apt install -y openssl libev4
-COPY app.exe .
-ENTRYPOINT /app.exe
+FROM ocaml/opam:alpine as build
+
+# Install system dependencies
+# Some of these are only needed for database and JavaScript stuff, but this is 
+# our build context, it won't impact our runtime environment.
+RUN sudo apk add --update \
+    make libffi-dev linux-headers m4 ncurses-dev perl pkgconf postgresql-dev \
+    musl-dev git pkgconfig gmp-dev libc-dev openblas-dev zlib-dev \
+    npm nodejs libev-dev 
```

Comment:
AFAIK, many of these are not necessary for an esy build. Have you confirmed that they are? See the instructions in the README.md for an approximate set of needed system dependencies.

Reply:
Right, many of these are not required, but they are needed for typical web applications (Postgres database, npm stack, etc.). Hopefully the benefit of not having to figure out which dependencies are missing in the build context outweight the added time to install the dependencies (a few seconds). Note that this doesn't have any impact on the produced Docker image as this is only the build context: the runtime context doesn't include any of these.

Reply:
There are more issues, like people will be looking at this example as a reference, not only using it as a starting point. They might be writing something completely different, and it will mislead them to include, for example, `postgresql-dev`, because it appears here as if it is an inherent dependency of the example.

For example, to reverse this case for a common scenario most people have probably experienced, if this was an Ubuntu container and it had a bunch of APT packages, but I was building an Alpine container, all these extra dependencies would cause me as a user to waste time looking up their Alpine equivalents &mdash even though many of the dependencies are ultimately not needed. This is very real with @ulrikstrid's suggestion to go the other way &mdash; translating Alpine packages back to Debian equivalents for `debian-slim`.

If I am in a more exotic environment, the amount of time wasted could be much greater, if I end up needlessly building from source, finding obscure versions due to some version incompatibitlies, and so on.

I don't think we can assume that these examples will be used primarily as starting points for development based on Alpine only, so I still suggest to minimize the system dependencies.

Reply:
Done! We end up with only `libev-dev openssl-dev` for Opam and `nodejs npm perl-utils m4 linux-headers` for Esy